components/formats-bsd/src/loci/formats/in/DicomReader.java

```diff
@@ -763,24 +763,22 @@ else if (isJPEG || isJP2K) {
 
         // all physical sizes were stored in mm, so must be converted to um
         if (pixelSizeX != null) {
-          Double sizeX = new Double(pixelSizeX);
-          sizeX *= 1000;
-          Length x = FormatTools.getPhysicalSizeX(sizeX);
+          Double physicalSizeX = new Double(pixelSizeX);
+          Length x = FormatTools.getPhysicalSizeX(physicalSizeX, "mm");
           if (x != null) {
             store.setPixelsPhysicalSizeX(x, i);
           }
         }
         if (pixelSizeY != null) {
-          Double sizeY = new Double(pixelSizeY);
-          sizeY *= 1000;
-          Length y = FormatTools.getPhysicalSizeY(sizeY);
+          Double physicalSizeY = new Double(pixelSizeY);
+          Length y = FormatTools.getPhysicalSizeY(physicalSizeY, "mm");
```

Comment:
Using the enums throughout the code (rather than string representations like `"mm"`) will make later searching/refactoring easier if it doesn't make the code too unwieldy now.